WinUIGallery/ControlPages/ScratchPadPage.xaml.cs

```diff
@@ -0,0 +1,575 @@
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls.Primitives;
+using Microsoft.UI.Xaml.Data;
+using Microsoft.UI.Xaml.Input;
+using Microsoft.UI.Xaml.Media;
+using Microsoft.UI.Xaml.Navigation;
+using Windows.Media.Capture.Frames;
+using Windows.Media.Capture;
+using Microsoft.UI.Xaml.Media.Imaging;
+using Windows.Media.MediaProperties;
+using Windows.Storage.Streams;
+using System.ComponentModel;
+using Microsoft.UI.Input;
+using Microsoft.UI.Xaml.Markup;
+using System.Xml;
+using Windows.Storage;
+
+namespace WinUIGallery.ControlPages
+{
+    public sealed partial class ScratchPadPage : Page
+    {
+        public ScratchPadPage()
+        {
+            this.InitializeComponent();
+
+            var xamlStr = ReadScratchPadXAMLinLocalSettings();
+
+            // If there is no stored XAML, load the default.
+            if (xamlStr == null || xamlStr.Trim().Length == 0)
+            {
+                xamlStr = GetDefaultScratchXAML();
+            }
+
+            m_oldText = xamlStr;
+            textbox.TextDocument.SetText(Microsoft.UI.Text.TextSetOptions.None, m_oldText);
+            var formatter = new XamlTextFormatter(textbox);
+            formatter.ApplyColors();
+
+            // Provide some initial instruction in the content area.
+            SetEmptyScratchPadContent();
+        }
+
+        private void SetEmptyScratchPadContent()
+        {
+            scratchPad.Content = new TextBlock()
+            {
+                Text = "Click the Load button to load the content below.",
+                HorizontalAlignment = HorizontalAlignment.Center,
+                VerticalAlignment = VerticalAlignment.Center,
+                TextWrapping = TextWrapping.Wrap
+            };
+        }
+
+        private string GetDefaultScratchXAML()
+        {
+            return
+@"<StackPanel  BorderThickness=""1"" BorderBrush=""Green"" CornerRadius=""4"" Padding=""3"">
+    <!-- Note: {x:Bind} is not supported in Scratch Pad. -->
+    <TextBlock>This is a sample TextBlock.</TextBlock>
+    <Button Content=""Click me!""/>
+
+    <!-- Note: Syntax highlighting updates on 'Load'. -->
+</StackPanel>";
+        }
+
+        public string ReadScratchPadXAMLinLocalSettings()
+        {
+            var appData = Windows.Storage.ApplicationData.Current;
+            if (appData.LocalSettings.Containers.ContainsKey("ScratchPad"))
+            {
+                var scratchPadContainer = appData.LocalSettings.CreateContainer("ScratchPad", Windows.Storage.ApplicationDataCreateDisposition.Existing);
+                if (scratchPadContainer != null && scratchPadContainer.Values.ContainsKey("ScratchPadXAML"))
+                {
+                    // String values are limited to to 4K characters. Use a composite value to support a longer string.
+                    var compositeStr = scratchPadContainer.Values["ScratchPadXAML"] as ApplicationDataCompositeValue;
+                    var xamlStr = "";

```

Comment:
consider [StringBuilder](https://learn.microsoft.com/en-us/dotnet/api/system.text.stringbuilder?view=net-8.0) for performance

Reply:
or better yet, use C#/WinRT string marshaling, which can create a fast-pass HSTRING (reference) for writing:
https://github.com/microsoft/CsWinRT/blob/master/docs/interop.md#strings

Reply:
Could use a StringBuilder, but this loop will typically run only once or twice, and only on page load or Reset. I think a StringBuilder isn't necessary in these conditions.